piff/polynomial_interp.py

```diff
@@ -0,0 +1,361 @@
+        is expecting this.
+
+        This is an internal method used during the fitting.
+        :param pos:        A numpy array of the u,v positions at which to build 
+                           the model
+        :param C:          A 2D matrix of polynomial coefficients in the form that
+                           the numpy polynomial form is expecting:
+                           p(x,y,c) = sum_{i,j} c_{ij} x^i y^j
+                           
+        :returns:          A numpy array of the calculated p_x(x)*p_y(y) where 
+                           the p functions are polynomials.
+
+        """
+        #Take the u and v components (x and y in the tangent plane)
+        #as our interpolants
+        u = pos[0]
+        v = pos[1]
+        #Call the appropriate function to generate the polynomial model.
+        #By default this is numpy.polyval. Note that despite appearances
+        #this is not a method call - function is a normal python attribute
+        #that happens to be a function.
+        f = self.function(u, v, C)
+        return f
+
+    def initialGuess(self, positions, parameter,parameter_index):
+        """Make an initial guess for a set of parameters
+        to use in the fit for your model. This is passed
+        to curve_fit as a starting point.
+
+        :param positions:  A list of positions ((u,v) in this case) of stars.
+        :param parameter:  A numpy array of the measured values of a parameter
+                           for each star
+                           
+        :returns:          A guess for the parameters. In this case a 2D matrix
+                           which is zero everywhere except for (0,0).  This should
+                           correspond to a flat function of the parameters with 
+                           value given by the mean.
+
+
+        """        
+        #We need a starting point for the fitter.
+        #Use a constant value over the whole field as 
+        #a reasonable guess.
+        n = self.orders[parameter_index]+1
+        C = numpy.zeros((n,n))
+        C[0,0] = parameter.mean()
+        return C
+
+
+
+    def solve(self, pos, vectors, logger=None):
+        """Solve for the interpolation coefficients given some data,
+        using the scipy.optimize.curve_fit routine, which uses Levenberg-Marquardt
+        to find the least-squares solution.
+
+        This currently assumes that our positions pos are just u and v.
+
+        :param pos:         A list of positions to use for the interpolation.
+        :param vectors:     A list of parameter vectors (numpy arrays) for each star.
+        :param logger:      A logger object for logging debug info. [default: None]
+        """
+        import scipy.optimize
+
+        #We will want to index things later, so useful
+        #to convert these to numpy arrays and transpose
+        #them to the order we need.
+        parameters = numpy.array(vectors).T
+        positions = numpy.array(pos).T
```

Comment:
It would be nice to log some information here.  Something like
```
if logger:
    logger.info("Fitting %d parameter vectors using polynomial type %s",len(vectors),self.poly_type)
```
Probably some extra detail at the `debug` level too would be helpful.